builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,206 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
-        </Web3Button>
-      );
-    }
-    ```
+const number = await readContract({
+      contract: contract,
+      method: "number",
+      params: [],
+    });
+```
 
 ### Connect Wallet {: #connect-wallet }
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
-
-The following example will show you how to use the `ConnectWallet` component. To go this route, you will need to specify the supported wallets and pass them to your provider.
-
-```javascript
-import {
-  ThirdwebProvider,
-  metamaskWallet,
-  coinbaseWallet,
-  walletConnectV1,
-  walletConnect,
-  safeWallet,
-  paperWallet,
-} from '@thirdweb-dev/react';
-
-function MyApp() {
+Next, let's customize the [Connect Button](https://portal.thirdweb.com/typescript/v5/react/components/ConnectButton){target=\_blank} to tailor it our desired wallets. You can add or remove wallets from the wallets array to change the options available to users. ThirdWeb also offers a [ConnectButton Playground](https://thirdweb.com/dashboard/connect/playground) to customize and view changes in real-time given the high degree of flexibility offered by the button. 
+
+```typescript title="App.tsx"
+import { ConnectButton } from "thirdweb/react";
+import { createWallet, inAppWallet } from "thirdweb/wallets";
+ 
+const wallets = [
+  inAppWallet(),
+  createWallet("io.metamask"),
+  createWallet("com.coinbase.wallet"),
+  createWallet("me.rainbow"),
+];
+ 
+function Example() {
   return (
-    <ThirdwebProvider
-      supportedWallets={[
-        metamaskWallet(),
-        coinbaseWallet(),
-        walletConnect({
-          projectId: 'INSERT_YOUR_PROJECT_ID', // optional
-        }),
-        walletConnectV1(),
-        safeWallet(),
-        paperWallet({
-          clientId: 'INSERT_YOUR_CLIENT_ID', // required
-        }),
-      ]}
-      activeChain={Moonbeam}
-    >
-      <App />
-    </ThirdwebProvider>
+    <div>
+      <ConnectButton client={client} wallets={wallets} />
+    </div>
   );
 }
 ```
 
-Next, you'll need to add in a connect wallet button to prompt end-users to login with any of the above supported wallets.
+## Deploy Application {: #deploy-application }
 
-```javascript
-import { ConnectWallet } from '@thirdweb-dev/react';
+Putting it all together, you can view the full code for the App.tsx file below:
 
-function App() {
-  return <ConnectWallet />;
-}
+??? code "View the complete App.tsx script"
+    ```typescript
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx'
+    ```
+
+
+And that's it! As a reminder, you can run your app locally with the following command:
```

Comment:
"And that's it" is confusing because this section is about deploying the application, but so far we've just reviewed the entire script